### ISSUE TITLE
vue/no-side-effects-in-computed-properties with composition api's computed

### ISSUE THREAD

**Issue by sapphi-red** [vuejs/eslint-plugin-vue#1393]
**What rule do you want to change?**
vue/no-side-effects-in-computed-properties

**Does this change cause the rule to produce more or fewer warnings?**
more warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```vue
<script>
import { defineComponent, ref, computed } from 'vue'

export default defineComponent({
  setup() {
    const arr = ref(['aaa', 'b', 'c'])
    const sorted = computed(() => arr.value.sort((a,b) => a.length - b.length)) // here arr is being mutated
    return { sorted }
  }
})
</script>
```

**What does the rule currently do for this code?**
no warning

**What will the rule do after it's changed?**
show warning like `computed` properties

**Additional context**

Actually, this new feature does not affect computed **properties**, so it might have to be a new rule.

**Comment by sapphi-red** [comment]
I think `vue/return-in-computed-property` and `vue/no-async-in-computed-properties` has the same thing with this.
Should I create another issue?
